eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/DbFeeProvider.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.blockchain.fee
+
+import fr.acinq.eclair.db.FeeratesDb
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * This provider first tries to retrieve the feerates from the a database acting as a cache.
+ *
+ * If no provider already exists for this provider's name, or if the feerates age exceeds the provided max age,
+ * then the provider's actual `getFeeRates` future is called ; subsequent result is saved to the database.
+ */
+class DbFeeProvider(db: FeeratesDb, provider: FeeProvider, maxAgeMillis: Long = 1.day.toMillis)(implicit ec: ExecutionContext) extends FeeProvider {
```

Comment:
Shouldn't we do this the other way around?
First try to get feerate from the provider. If it fails, fetch from the DB. If the DB entry is obsolete, fail.
This would ensure we have more fresh entries in the DB, because with the current code we will only try to fetch from the provider once the db entry is obsolete, so if that single call fails we're screwed...

Reply:
If the main goal is the faster start-up on android, it makes sense to first query the DB, but then even if the DB call succeeds I would do the call to the provider to ensure we have a more recent entry in the DB for next time?

Reply:
In fact I changed this PR with 8651b04, this provider is not a cache anymore, the value in database will only be used once, and then it falls back to the wrapped provider. The only point of this PR is to have a fast startup, which happens only once, so a caching system is not useful.